restapi.py

```diff
@@ -55,100 +73,302 @@ class Advance(Resource):
+                    
+            return {'message':'failure','error':'invalid json input'}        
+        
         u = parser_advance.parse_args()
-        y = case.advance(u)
-        return y
+        
+        app.logger.info("Receiving a new advance request: {}".format(u)) 
+                
+        result = case.advance(u)
+        
+        if result['message'] == 'success':
+
+            app.logger.info("Advanced the simulation")
+        
+        else:
+        
+            app.logger.error("Fail to advanced the simulation: {}".format(result['error']))   
+                    
+        return result        
 
 class Initialize(Resource):
     '''Interface to initialize the test case simulation.'''
     
     def put(self):
         '''PUT request to initialize the test.'''
+        
+        parse = request.get_json(silent = True)
+        
+        if parse is None:
+        
+            app.logger.error("Receiving a invalid initialize request") 
+                    
+            return {'message':'failure','error':'invalid json input'}
+            
         args = parser_initialize.parse_args()
-        start_time = float(args['start_time'])
-        warmup_period = float(args['warmup_period'])
-        result = case.initialize(start_time,warmup_period)      
+        
+        app.logger.info("Receiving a new initialize request: {}".format(args)) 
+        
+        try: 
+        
+            start_time = float(args['start_time'])
+            
+            warmup_period = float(args['warmup_period'])
+            
+        except TypeError:
+        
+            app.logger.error("Receiving insufficient inputs for a initialize request")
+        
+            return {'message':'failure','error':'insufficient inputs'}
+             
+        except ValueError:
+        
+            app.logger.error("Receiving a invalid initialize request") 
+        
+            return {'message':'failure','error':'invalid input'}               
+            
+        result = case.initialize(start_time,warmup_period)  
+
+        if result['message'] == 'success':
+
+            app.logger.info("Reset the simulation start time to: {}".format(start_time)) 
+        
+        else:
+        
+            app.logger.error("Fail to initialize the simulation:{}".format(result['error']))                          
+        
         return result
 
 class Step(Resource):
     '''Interface to test case simulation step size.'''
     
     def get(self):
         '''GET request to receive current simulation step in seconds.'''
-        step = case.get_step()
-        return step
+                
+        app.logger.info("Receiving a new query for step") 
+        
+        try:
+        
+            step = case.get_step()
+            
+        except Exception as e:
+        
+            app.logger.error("Fail to return the simulation step:{}".format(e))
+        
+            return {'message':'failure','error':e,'result':None} 
```

Comment:
Should 'message', 'error', 'result', be standard and ``None`` given as value where applicable, as is done in this line?  Other returns in proposed code have only a subset of these keys.